draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -939,26 +939,30 @@ following conditions holds:
 
 ## "Same-site" and "cross-site" Requests  {#same-site-requests}
 
-A request is "same-site" if its target's URI's origin's registrable domain
-is an exact match for the request's client's "site for cookies", or if the
-request has no client. The request is otherwise "cross-site".
+Two origins, A and B, are considered same-site if the following algorithm 
+returns true:
+
+1.  If A and B are both scheme/host/port triples:
 
-For a given request ("request"), the following algorithm returns `same-site` or
-`cross-site`:
+    1.  If A's scheme does not equal B's scheme, return false.
 
-1.  If `request`'s client is `null`, return `same-site`.
+    2.  Let hostA be A's host, and hostB be B's host.
 
-    Note that this is the case for navigation triggered by the user directly
-    (e.g. by typing directly into a user agent's address bar).
+    3.  If hostA equals hostB and hostA's registrable domain is null, return 
+        true.
 
-2.  Let `site` be `request`'s client's "site for cookies" (as defined in the
-    following sections).
+    4.  If hostA's registrable domain equals hostB's registrable domain and is 
+        non-null, return true.
 
-3.  Let `target` be the registrable domain of `request`'s current url.
+2.  If A and B are both the same globally unique identifier, return true.
 
-4.  If `site` is an exact match for `target`, return `same-site`.
+3.  Return false.
 
-5.  Return `cross-site`.
+Note: The port component of the origins is not considered.
+
+A request is "same-site" if its target's URI's origin
+is same-site with the request's client's "site for cookies", or if the
+request has no client. The request is otherwise "cross-site".
 
 The request's client's "site for cookies" is calculated depending upon its
```

Comment:
Maybe spell out explicitly that the site for cookies is an origin.

Reply:
The next sentence refers to determining the "site for cookies" depending on the type (documents & dedicated workers: origin, shared workers: maybe origin, maybe unique identifier, service worker: service worker's URI origin). 

So I think we're fine leaving it as is.

Reply:
Actually, I see what you mean now.